pilot/cmd/pilot-agent/main.go

```diff
@@ -68,6 +70,7 @@ const (
 // TODO: Move most of this to pkg.
 
 var (
+	platform           string
```

Comment:
We generally don't use 'is k8s' or platform specific code. 
Add a specific setting for whatever you need- like 'longLivedToken' ( please don't call it 'always valid' - nothing is always valid, even k8s tokens can be revoked )

Reply:
This is a very large change that greatly impact Env WG - it we do want to make such a fundamental changes, a RFC and discussion would be better. 

There are many implications in terms of config - what is a platform ? GCP ? VM ? GKE ? 

Also - this is not needed, it is simpler and cleaner - and more consistent with rest of istio - to have an explicit plugin